javascript/modules/client/src/client-erc20.ts

```diff
@@ -241,3 +260,79 @@ function unwrapWithdrawParams(
     params.reference ?? "",
   ];
 }
+
+/// HELPER FUNCTIONS TODO: move to utils?
+function setProposalStatus(proposals: IErc20VotingProposal[]) {
+  const now = new Date();
+
+  for (let proposal of proposals) {
+    if (proposal.startDate >= now) {
+      proposal.status = ProposalStatus.PENDING;
+    } else if (proposal.endDate >= now) {
+      proposal.status = ProposalStatus.ACTIVE;
+    } else if (proposal.executed) {
+      proposal.status = ProposalStatus.EXECUTED;
+    } else if (proposal.yea && proposal.nay && proposal.yea > proposal.nay) {
+      proposal.status = ProposalStatus.SUCCEEDED;
+    } else {
+      proposal.status = ProposalStatus.DEFEATED;
+    }
+  }
+}
+
+/// MOCK DATA FOR PROPOSALS
+const dateWithinThisYear = new Date(
+  new Date().setFullYear(new Date().getFullYear() - 1)
+).getTime();
+
+const startDate = new Date(
+  dateWithinThisYear + Math.random() * (Date.now() - dateWithinThisYear)
+);
+
+const MOCK_PROPOSALS = [
+  {
+    id: "0x56fb7bd9491ff76f2eda54724c84c8b87a5a5fd7_0x0",
+    daoAddress: "0x56fb7bd9491ff76f2eda54724c84c8b87a5a5fd7",
+    daoName: "DAO 1",
+    creator: "0x1234...",
+
+    startDate,
+    endDate: new Date(startDate.getTime() + 7200000),
+    createdAt: new Date(dateWithinThisYear),
+
+    title: "New Founding for Lorex Lab SubDao",
+    summary:
+      "As most community members know, Aragon has strived to deploy its products to more cost-efficient blockchain networks to facilitate interaction.",
+    proposal: "<h1>This is the super important proposal body<h1>",
+    resources: [{ url: "https://example.com", description: "Example" }],
+
+    executed: false,
+    status: ProposalStatus.PENDING,
+
+    voteId: "0",
+    token: {
+      address: "0x1234...",
+      name: "DAO Token",
+      symbol: "DAO",
+      decimals: 18,
+    },
+
+    yea: 3,
+    nay: 1,
+    abstain: 2,
+
+    open: true,
+    participationRequiredPct: 30,
+    supportRequiredPct: 52,
+    votingPower: 135,
+
+    voters: [
+      { id: "0x1334...", voterState: VoteOptions.YEA, weight: 1 },
```

Comment:
- Does `id` refer to the address of the voter?
- Does `voterState` refer to the vote value?